src/Columns/ColumnString.h

```diff
@@ -62,6 +62,8 @@ class ColumnString final : public COWHelper<IColumn, ColumnString>
     ColumnString(const ColumnString & src);
 
 public:
+    ~ColumnString() override = default;
```

Comment:
This change does not make sense, most likely misinterpreted.

Reply:
On my end, the static analyzer reported a violation of the `cppcoreguidelines-virtual-class-destructor` check. I can't seem to reproduce this using `clang-tidy-14`, even though this rule isn't disabled in the project's configuration file. If you wish I can revert this part, but I'd recommend keeping it because it ensures that the base class does have a virtual destructor via `override` and doesn't leave it to chance.

What's your opinion on this?

Reply:
The base class (`IColumn`) is an interface class with pure virtual methods but no member variables. Imho, the risk of slicing / mem leaks during destruction is negligible.

I also find it strange that the other `IColumn` subclasses have no virtual dtor and that clang-tidy-14 doesn't complain about this.

Suggest removing this part as you propose.